### PR TITLE
refactor: remove unused libraries

### PR DESCRIPTION
**Pull request by rosariopfernandes**
The initial idea here was to remove the legacy support library, since it seems like it is not being used anywhere in the code. But then I noticed that if we use the Material Components Theme, we might be able to remove some other libraries as well, such as `androidx.cardview:cardview:1.0.0`.

I was wondering if this is something we're interested in? If it is, I can do it to the remaining samples  on this same PR.

### CODE REVIEW

File: functions/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -20,8 +20,10 @@ specific language governing permissions and limitations under the License.
         android:theme="@style/AppTheme">
 
 
-        <activity android:name=".java.MainActivity"/>
-        <activity android:name=".kotlin.MainActivity"/>
+        <activity android:name=".java.MainActivity"
+            android:theme="@style/MaterialAppTheme"/>
```
Comment (samtstern) [inline functions/app/src/main/AndroidManifest.xml]: Is there a reason we can't put this `theme` on the `application` tag like the previous one?

File: functions/app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -14,15 +14,15 @@ specific language governing permissions and limitations under the License.
...
-    android:paddingBottom="@dimen/activity_vertical_margin"
-    android:paddingLeft="@dimen/activity_horizontal_margin"
-    android:paddingRight="@dimen/activity_horizontal_margin"
-    android:paddingTop="@dimen/activity_vertical_margin"
     tools:context="com.google.samples.quickstart.functions.java.MainActivity">
 
     <LinearLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
+        android:paddingBottom="@dimen/activity_vertical_margin"
```
Comment (samtstern) [inline functions/app/src/main/res/layout/activity_main.xml]: Is this change necessary with the new library/theme or just something you chose to do?

File: functions/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (rosariopfernandes) [reply functions/app/src/main/AndroidManifest.xml]: That's because I wanted to apply it to that Activity only, keeping the ChooserActivity with the previous theme. But if we go forward with the switch, it will be moved to the application tag.

File: functions/app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (rosariopfernandes) [reply functions/app/src/main/res/layout/activity_main.xml]: Oh, my bad, forgot to mention this.

It is something I chose to do. With the previous layout we could see the padding around the scrolling area. Here's a screenshot:

device-2019-10-25-190023 resized

Comment (samtstern) [reply functions/app/src/main/res/layout/activity_main.xml]: Ah yeah ok, your fix :+1:

**Comment by rosariopfernandes** [comment]
## Progress Tracking:
- [x] AdMob
- [x] Analytics
- [x] App Indexing
- [ ] Authentication
- [ ] Remote Config
- [ ] Crashlytics
- [ ] Realtime Database
- [ ] Dynamic Links
- [x] Cloud Firestore
- [x] Cloud Functions
- [x] In-App Messaging
- [ ] Cloud Messaging
- [ ] ML Kit
- [ ] ML Kit Language ID
- [ ] ML Kit Smart Reply
- [ ] ML Kit Translate Text
- [x] Performance Monitor
- [ ] Cloud Storage

File: mlkit-langid/app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -25,7 +25,7 @@
             android:hint="@string/help_text" />
     </com.google.android.material.textfield.TextInputLayout>
 
-    <androidx.constraintlayout.Guideline
+    <androidx.constraintlayout.widget.Guideline
```
Comment (rosariopfernandes) [inline mlkit-langid/app/src/main/res/layout/activity_main.xml]: The previous implementation was causing the app to crash on start. Seems like there was a breaking change on the latest version of ConstraintLayout. Found this solution on [StackOverflow](https://stackoverflow.com/a/52903072/5861618)

**Comment by samtstern** [comment]
@rosariopfernandes just let me know when you're ready for review!  Doesn't have to be 100% done you can always do RTDB and MLKit in a follow-up

**Comment by rosariopfernandes** [comment]
Thanks @samtstern , but I've already done both of them.
This is ready for review now.

File: auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/ChooserActivity.kt [hunk 4]
```diff
@@ -34,7 +34,7 @@ class ChooserActivity : AppCompatActivity(), AdapterView.OnItemClickListener {
         setContentView(R.layout.activity_chooser)
 
         // Set up Adapter
-        val adapter = MyArrayAdapter(this, android.R.layout.simple_list_item_2, CLASSES)
+        val adapter = MyArrayAdapter(this, android.R.layout.simple_list_item_2, CLASSES as Array<Class<*>>)
```
Comment (rosariopfernandes) [inline auth/app/src/main/java/com/google/firebase/quickstart/auth/kotlin/ChooserActivity.kt]: This is unrelated, but Android Studio was showing an error saying it expected `Array<Class<*>>`, but found  `Array<out Class<out Any>>`.

File: analytics/app/src/main/java/com/google/firebase/quickstart/analytics/java/MainActivity.java [hunk 5]
```diff
@@ -266,8 +267,9 @@ private void recordScreenView() {
 
         private final ImageInfo[] infos;
 
+        @SuppressLint("WrongConstant")
         public ImagePagerAdapter(FragmentManager fm, ImageInfo[] infos) {
-            super(fm);
+            super(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT);
```
Comment (rosariopfernandes) [inline analytics/app/src/main/java/com/google/firebase/quickstart/analytics/java/MainActivity.java]: Added the constant to match its Kotlin equivalent

File: analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt [hunk 6]
```diff
@@ -218,6 +219,7 @@ class MainActivity : AppCompatActivity() {
      * A [FragmentPagerAdapter] that returns a fragment corresponding to
      * one of the sections/tabs/pages.
      */
+    @SuppressLint("WrongConstant")
```
Comment (rosariopfernandes) [inline analytics/app/src/main/java/com/google/firebase/quickstart/analytics/kotlin/MainActivity.kt]: Added this to ignore Lint message:
> Must be one of: androidx.fragment.app.FragmentPagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT or androidx.fragment.app.FragmentPagerAdapter.BEHAVIOR_SET_USER_VISIBLE_HINT

**Comment by samtstern** [comment]
@rosariopfernandes thank you *SO* much for doing all this!  LGTM if it builds.
